src/torchcodec/decoders/_core/CudaDevice.cpp

```diff
@@ -256,4 +256,36 @@ void convertAVFrameToDecodedOutputOnCuda(
           << " took: " << duration.count() << "us" << std::endl;
 }
 
+// inspired by https://github.com/FFmpeg/FFmpeg/commit/ad67ea9
+void forceCudaCodec(
+    const torch::Device& device,
+    AVCodecPtr* codec,
+    const AVCodecID& codecId) {
+  if (device.type() != torch::kCUDA) {
```

Comment:
Replace this check with `throwErrorIfNonCudaDevice(device)`. That's the convention used by the other functions in this file, and it also enforces that calling this function in a non-CUDA context is an error.